### ISSUE TITLE
just a general question from newbie. any help highly appreciated :)

### ISSUE THREAD

**Issue by ooodragon94** [open-mmlab/mmsegmentation#1282]
hi
thanks for your great repo

I have some questions
I really appreciate any help or intuition on where I might be wrong.
I am currently using a custom data to train a network.
I have saved my masks on ann_dir/{xxx}.png with class labels within each pixels.

It seems like the validation accuracy won't increase.

The first problem I found was that it replaces my background class (whose index is 0) to 255 and it is ignored while calculating cross entropy. So I manually fixed it by changing it back to 0, since I don't know I can fix this with configuration files.

The second problem I'm facing is that, classes with dominating frequencies has low accuracy. This is weird since my baseline code with baseline architecture is overconfident on those classes. I'm beginning to think that I am feeding wrong classes for my custom data. However, I am still struggling to found out whether this conjecture is right or wrong...

Any intuitions, methods or ideas are welcome!

Thank you very much for your help.

**Comment by ooodragon94** [comment]
very strange but it seems like dataset loads label with (class_idx - 1)
I don't know why it does it.

Can anyone explain? And can anyone tell me the preferred way (or the mmseg way) to

- count background class (class idx = 0), which is changed to 255 while preprocessing, for cross entropy
- disable this "class index - 1" when calling dataset __get_item__ method

**Comment by sainivedh19pt** [comment]
Hi @ooodragon94 ,

The annotation or gt loading semantic masks starts [here](https://github.com/open-mmlab/mmsegmentation/blob/0934a57f4f42a2c16da559bab4be75503068d0c7/mmseg/datasets/pipelines/loading.py#L91).

**Comment by ooodragon94** [comment]
@sainivedh19pt Thank you so much for your thoughts!
I will have better time finding out which part is wrong!
